core/include/gnuradio-4.0/Port.hpp

```diff
@@ -365,13 +371,29 @@ struct Port {
                         break;
                     }
                 }
-                std::ignore = tags.tryConsume(tagsToConsume); // this is a default fallback: tags may be both explicitely and implicitly consumed
+                std::ignore = tags.tryConsume(tagsToConsume); // this is a default fallback: tags may be both explicitly and implicitly consumed
             }
         }
     };
-
     static_assert(ConsumablePortSpan<ConsumablePortInputRange<gr::SpanReleasePolicy::ProcessAll>>);
 
+    template<SpanReleasePolicy spanReleasePolicy>
+    using WriterSpanType = decltype(std::declval<WriterType>().template tryReserve<spanReleasePolicy>(1UZ));
+
+    template<SpanReleasePolicy spanReleasePolicy>
+    struct PublishablePortOutputRange : public WriterSpanType<spanReleasePolicy> {
+        ThisPortFullType* _port;
+        // explicit PublishablePortOutputRange(WriterSpanType<spanReleasePolicy>& span) : WriterSpanType<spanReleasePolicy>(span) {};
+
+        template<typename TSpan>
+        explicit PublishablePortOutputRange(TSpan&& span, ThisPortFullType* port) : WriterSpanType<spanReleasePolicy>(std::forward<TSpan>(span)), _port(port){};
+
+        ~PublishablePortOutputRange() = default;
+
+        void publishTag(property_map& tagData, Tag::signed_index_type index) { _port->publishTag(tagData, index); }
```

Comment:
Maybe rename `index` to `tagOffset` to better express the intention that this is the index/offset w.r.t. the initial sample that is presented to the user.